src/Server/WPHelper.php

```diff
@@ -46,22 +50,79 @@ public function parseRequestParams( $method, array $bodyParams, array $queryPara
 		 *
-		 * @param array $data An array containing the pieces of the data of the GraphQL request
-		 * @param array $request_context An array containing the both body and query params
+		 * @param array  $parsed_body_params An array containing the pieces of the data of the GraphQL request.
+		 * @param array  $request_context    An array containing the both body and query params.
+		 * @param string $method             The method of the request (GET, POST, etc).
 		 */
+		$parsed_body_params = apply_filters( 'graphql_request_data', $parsed_body_params, $request_context, $method );
+
+		// In GET requests there cannot be any body params so it's empty.
 		if ( 'GET' === $method ) {
-			$parsed_query_params = apply_filters( 'graphql_request_data', $parsed_query_params, $request_context );
-			// In GET requests there cannot be any body params so it's empty.
 			return parent::parseRequestParams( $method, [], $parsed_query_params );
 		}
 
 		// In POST requests the query params are ignored by default but users can
 		// merge them into the body params manually using the $request_context if
 		// needed.
-		$parsed_body_params = apply_filters( 'graphql_request_data', $parsed_body_params, $request_context );
 		return parent::parseRequestParams( $method, $parsed_body_params, [] );
 	}
 
+	/**
+	 * Process multi part requests.
+	 *
+	 * @param string $method             The method of the request (GET, POST, etc).
+	 * @param array  $parsed_body_params An array containing the pieces of the data of the GraphQL request
+	 * @return mixed
+	 */
+	private function process_multi_part_requests( $method, array $parsed_body_params ) {
+		$contentType = wp_unslash( $_SERVER['CONTENT_TYPE'] ?? null ); // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+
+		// Bail early.
+		if ( 'POST' !== $method || stripos( $contentType, 'multipart/form-data' ) === false ) {
+			return $parsed_body_params;
+		}
+
+		if ( empty( $parsed_body_params['map'] ) ) {
+			throw new RequestError( __( 'The request must define a `map`', 'wp-graphql' ) );
+		}
+
+		$map    = $this->decode_json( $parsed_body_params['map'] );
+		$result = $this->decode_json( $parsed_body_params['operations'] );
+
+		foreach ( $map as $fileKey => $locations ) {
+			$items = &$result;
+
+			foreach ( $locations as $location ) {
+				foreach ( explode( '.', $location ) as $key ) {
+					if ( ! isset( $items[ $key ] ) || ! is_array( $items[ $key ] ) ) {
+						$items[ $key ] = [];
+					}
+
+					$items = &$items[ $key ];
+				}
+
+				$items = $_FILES[ $fileKey ]; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotValidated,WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+			}
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Decode json.
+	 *
+	 * @param mixed $json Json.
+	 * @return mixed
+	 */
+	private function decode_json( $json ) {
+
+		if ( ! is_string( $json ) ) {
+			return $json;
+		}
+
+		return json_decode( stripslashes( $json ), true );
```

Comment:
We should probably check `json_last_error()`, instead of just assuming we're going to get an array.

Reply:
Good idea!